casepro/msgs/models.py

```diff
@@ -132,6 +133,125 @@ def __str__(self):
 
 
 @python_2_unicode_compatible
+class Language(models.Model):
+    """
+    Languages used, defined by the ISO 639-3 language code combined with the location code, e.g. eng_UK
+    """
+    org = models.ForeignKey(Org, verbose_name=_("Organization"), related_name='languages')
+
+    code = models.CharField(max_length=6)  # e.g. eng_UK
+
+    name = models.CharField(max_length=100, null=True, blank=True)  # e.g. English
+
+    location = models.CharField(max_length=100, null=True, blank=True)  # e.g. United Kingdom
+
+    @classmethod
+    def search(cls, org, user, search):
+        """
+        Search for Languages
+        """
+        name = search.get('name')
+        location = search.get('location')
+
+        queryset = Language.objects.all()
+
+        # Name filtering
+        if name:
+            queryset = queryset.filter(name__icontains=name)
+
+        # Location filtering
+        if location:
+            queryset = queryset.filter(location__icontains=location)
+
+        return queryset.order_by('code')
+
+    def as_json(self):
```

Comment:
Follow pattern in other models where the arg `full` determines whether this should return the full representation, or just a short id/name representation used as a reference from another object.